python-sdk/nuscenes/predict/input_representation/agents.py

```diff
@@ -0,0 +1,279 @@
+    return new_rgb
+
+def default_colors(category_name: str) -> Tuple[int, int, int]:
+    """
+    Maps a category name to an rgb color (unfaded).
+    :param category_name: Name of object category for the annotation.
+    :return: Tuple representing rgb color.
+    """
+
+    if 'vehicle' in category_name:
+        return (255, 255, 0) # yellow
+    elif 'object' in category_name:
+        return (204, 0, 204) # violet
+    elif 'human' in category_name:
+        return (255, 153, 51) # orange
+    else:
+        raise ValueError(f"Cannot map {category_name} to a color.")
+
+
+def draw_agent_boxes(center_agent_annotation: Dict[str, Any],
+                     center_agent_pixels: Tuple[int, int],
+                     agent_history: History,
+                     base_image: np.ndarray,
+                     get_color: Callable[[str], Tuple[int, int, int]],
+                     resolution: float = 0.1) -> None:
+    """
+    Draws past sequence of agent boxes on the image.
+    :param center_agent_annotation: Annotation record for the agent
+        that is in the center of the image.
+    :param center_agent_pixel: Pixel location of the agent in the
+        center of the image.
+    :param agent_history: History for all agents in the scene.
+    :param base_image: Image to draw the agents in.
+    :param get_color: Mapping from category_name to RGB tuple.
+    :param resolution: Size of the image in pixels / meter.
+    :return: None.
+    """
+
+    agent_x, agent_y = center_agent_annotation['translation'][:2]
+
+    for instance_token, annotations in agent_history.items():
+
+        num_points = len(annotations)
+
+        for i, annotation in enumerate(annotations):
+
+            box = get_track_box(annotation, (agent_x, agent_y), center_agent_pixels, resolution)
+
+            if instance_token == center_agent_annotation['instance_token']:
+                color = (255, 0, 0)
+            else:
+                color = get_color(annotation['category_name'])
+
+            # Don't fade the colors if there is no history
+            if num_points > 1:
+                color = fade_color(color, i, num_points - 1)
+
+            cv2.fillPoly(base_image, pts=[np.int0(box)], color=color)
+
+
+def get_rotation_matrix(image_shape: Tuple[int, int, int], yaw_in_radians: float) -> np.ndarray:
+    """
+    Gets a rotation matrix to rotate a three channel image.
+    :param image_shape: (Length, width, n_channels)
+    :param yaw_in_radians: Angle to rotate the image by.
+    """
+
+    rotation_in_degrees = angle_of_rotation(yaw_in_radians) * 180 / np.pi
+
+    return cv2.getRotationMatrix2D((image_shape[1]/2, image_shape[0]/2), rotation_in_degrees, 1)
+
+
+class AgentBoxesWithFadedHistory(AgentRepresentation):
+    """
+    Represents the past sequence of agent states as a three-channel
+    image with faded 2d boxes.
+    """
+
+    def __init__(self, helper: PredictHelper,
+                 seconds_of_history: int = 2,
```

Comment:
Weird that seconds has int units, while frequency is float. Shouldn't both be float?

Also, may want to be more explicit on the units of resolution w/ a comment.

Reply:
Added the comment!